roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/Sink.java

```diff
@@ -26,12 +29,22 @@ public double getLength() {
 
     @Override
     public void addCar(ICar car) {
+        if (carQueue == null) {
+            throw new IllegalStateException("carQueue in section cannot be null");
+        }
+
+        carQueue.addLast(car);
         this.carCounter++;
         CarController.removeCarMapping(car);
     }
 
     @Override
     public ICar getFirstCar() {
+        final List<ICar> carQueue = getCarQueue();
```

Comment:
this stuff i see is only for the test you wrote but has nothing to do with the usage of a sink..
i would propose to move this stuff into the test/helper method in the test class which collects the cars and then returns the first car